### PR TITLE
Post 2.50.0 Streamline release guide improvements

### PR DESCRIPTION
**Pull request by lostluck**
This is a larger cleanup of the release guide to clarify timings and positioning of the broader steps.

Larger emphasis has been put on the Release Manager's role prior to the release cut. If issues are addressed in the main branch before the cut, they do not need to be addressed afterwards. The release manager is empowered to do so.

Clarify some timing around the begining and end of the release by calling out the preparation steps should be performed within the week prior to the release cut. Also call out that release finalization tasks can be done within a few hours of the release being completed.

While the guide isn't materially shorter, maintaining the simpler structure as in the overview will make it less overwheling to release managers, since it's clearer when the various subtasks apply, vs the previous "13" steps. This should clarify the "build release candidate + vote + fix" loop that is the bulk of a release manager's tasks. I believe this will help get to release candidates sooner.

Add a new section "Post Release Tasks" which is where the Playground update task lives. While it's part of the release tasks, it doesn't block the release overall, so it felt weird structurally to have it as important as "Build a Release Candidate".

Various minor typos and fixes.

Rendered version should become available here:

http://apache-beam-website-pull-requests.storage.googleapis.com/28287/contribute/release-guide/index.html

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by lostluck** [comment]
R: @damccorm @kennknowles @jrmccluskey @abacn

File: website/www/site/content/en/contribute/release-guide.md [hunk 1]
```diff
@@ -386,55 +447,18 @@ For each of the failing tests create a GitHub Issue with the following propertie
...
-  Please consider discussing this with stakeholders and the dev@ mailing list, as appropriate.
-* If the issue has not been resolved and it is not acceptable to release until it is fixed, the release cannot proceed.
-  Instead, work with the Beam community to resolve the issue.
-
-If there is a bug found in the RC creation process/tools, those issues should be considered high priority and fixed in 7 days.
-
 ### Review cherry-picks
 
+The release manager is empowered to triage issues, and accept or regject cherry-picks to the release branch.
+Cherry picks are necessary if there are outstanding issues at time of the release branche cut, or issues were found in verification.
```
Comment (damccorm) [inline website/www/site/content/en/contribute/release-guide.md:457]:
```suggestion
Cherry picks are necessary if there are outstanding issues at time of the release branch cut, or issues were found in verification.
```

File: website/www/site/content/en/contribute/release-guide.md [hunk 2]
```diff
@@ -167,101 +228,103 @@ Configure access to the [Apache Nexus repository](https://repository.apache.org/
...
 
-#### Login to DockerHub
-If you are a member of the [`beam` DockerHub team](https://hub.docker.com/orgs/apache/teams/beam), run following command manually.
-It will ask you to input your DockerHub ID and password if authorization info cannot be found from ~/.docker/config.json file.
+#### Update the Java BOM
 
-```
-docker login docker.io
-```
+TODO(lostluck): Find instructions or associated issue/PRs (go/beam-issue/#####)
```
[the remark this answers is not in the document]
Comment (lostluck) [reply website/www/site/content/en/contribute/release-guide.md:253]: Ack. Contacting Bruno and Reuven who appear to have done this recently regarding the accuracy of the instruction on the wiki.

[the remark this answers is not in the document]
Comment (lostluck) [reply website/www/site/content/en/contribute/release-guide.md:253]: Filed https://github.com/apache/beam/issues/28379 for now.

**Comment by lostluck** [comment]
PTAL

**Review by damccorm — APPROVED** [review]
LGTM

**Review by kennknowles — APPROVED** [review]
I hate hate HATE markdown that is only readable when rendered (aka using a sequence of 1. 1. 1. 1.). But you didn't start the fire, for the most part...

**Comment by lostluck** [comment]
> I hate hate HATE markdown that is only readable when rendered (aka using a sequence of 1. 1. 1. 1.). But you didn't start the fire, for the most part...

I'm right there with you, but I'd make that a separate change across the whole file rather than adjust the untouched lists here, or only partially convert it.

**Pull request merged**